provisioners.proto

```diff
@@ -158,3 +159,17 @@ message SCEPProvisioner {
 message NebulaProvisioner {
 	repeated bytes roots = 1;
 }
+
+message Webhook {
+	string name = 1;
+	string url = 2;
+	enum Kind {
+		NO_KIND = 0;
+		ENRICHING = 1;
+	}
+	Kind kind = 3;
+	string secret = 4;
+	string bearer_token = 5;
+	string username = 6;
+	string password = 7;
```

Comment:
Although it makes the code a little bit more complicated, it might be better to use `oneof` here and have different types of authentication: BearerToken, SharedSecret, BasicAuth, ... 

See how the ProvisionerDetails are done.

Reply:
https://developers.google.com/protocol-buffers/docs/proto3#oneof